components/avatar/avatar.vue

```diff
@@ -125,33 +147,94 @@ export default {
   },
 
   methods: {
-    init () {
-      const firstChild = this.$el.firstChild;
-      if (firstChild) {
-        this.setKind(firstChild);
+    async init () {
+      const firstChild = this.$slots.default[0]?.elm || this.$slots.default[0];
+      const iconChild = this.$slots.default[0]?.componentOptions?.tag === 'dt-icon';
+
+      if (firstChild || iconChild) {
+        if (iconChild) {
+          this.kind = 'icon';
+        } else {
+          this.setKind(firstChild);
+        }
+
+        await this.$nextTick();
+        const childEl = this.$el;
+
         if (this.kind === 'image') {
-          firstChild.classList.add('d-avatar__image');
+          childEl.firstChild.classList.add('d-avatar__image');
+          // TO DO: remove attached event listener
+          childEl.firstChild.addEventListener('error', () => {
+            this.kind = 'initials';
+            this.imageLoadedSuccessfully = false;
+            childEl.firstChild.classList.remove('d-avatar__image');
```

Comment:
@braddialpad it's pending to fix the removal of the class when it falls back to the initials when the image couldn't be loaded. It's removing the first time with the listener but when the change is applied in the DOM, the `init` method is executed again due to the call in the `updated` hook so the class is applied again. If we add the class just when the image is successfully loaded, that implies that the size will be incorrect until it's fully loaded.